### DISCUSSION TITLE
Improving ManyToManyField with relational databse

### DISCUSSION THREAD

**Discussion by danielcatao** [bhch/django-jsonform#179] [Ideas]
![ManyToManyExample](https://github.com/user-attachments/assets/06336f86-22d3-4a53-b1af-29b191cce424)

Relational databases handle a manyToMany relation with a pivot table (ex. people_banks). 

A JSON representation of the same relationship could be: `[{"people_id":1, "bank_id":2,"account_type":"gold" }]`

A django-jsonform view of this case is: 

![Captura de tela 2024-11-14 220612](https://github.com/user-attachments/assets/f2006b2e-5c33-460b-b7ef-a6060e7d6d40)

Today the default Django Admin closest way to handle this N:M diagram is: 

![Captura de tela 2024-11-14 221822](https://github.com/user-attachments/assets/1a52ed13-a122-4e22-acd6-c5422fa25f28)

However, this limit the use of the pivot attribute account_type.

that said... my question/idea is: **Can we improve django-jsonform and consequently django admin itself to handle relation manyToMany better than the actual component "select multiple"? And if yes than how?**

**Comment by danielcatao** [comment]
Perhaps `People -> PeopleBanks <- Banks` wasn´t a good example, please, think of `Product -> OrderProduct <- Order` with OrderProduct having `product_price` attr.

**Comment by bhch** [comment]
I can see that selecting **Client** and **Bank** again and again can get tiresome.

Here's my solution: 

 1. Instead of sending an empty **Accounts** array from the server, you send the initial data array with all the **Client**–**Bank** combinations as items. This way django-jsonform will display all the items with Client and Bank data pre-filled for you.
 2. In the schema, make the **Select** input `readonly`.
 3. Set array's `min_items` and `max_items` to the total items in the list to prevent removing or adding items.
 4. Then you will only have to fill the **Account type** field.

**Reply by danielcatao** [reply]
I dont know if thats the best way because in a case of 1 milion records it will cost a lot of computing just to give options. 

I think that maybe a parser from json scheme onsave to convert to SQL insert would be better. 

Maybe can django-jsonform identify that it is being used by a manyToManyField? 

Ps: I'm sorry if I am not being clear. I'm coming from PHP and only learned python last week. I will dive into the code and figure out a better propose. I'm sorry for my english too. I'm from Brazil.

**Reply by danielcatao** [reply]
Perhaps the form can turn into a table too to save some screen space like it is the component Matrix from Laravel Orchid.

**Comment by danielcatao** [comment]
https://docs.djangoproject.com/en/5.0/ref/contrib/admin/#working-with-many-to-many-models (admin.TabularInline) did the job.. 

Thank you for the attention @bhch
